File: LC/python/Best Time to Buy and Sell Stock.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        if len(prices) == 0 :
            return 0
            
        minValue = float("inf")
        maxProfit = 0

        for i in range(len(prices)):
            if prices[i] < minValue:
                minValue = prices[i]
            if maxProfit < prices[i] - minValue:
                maxProfit = prices[i] - minValue
        return maxProfit
        

File: LC/java/Best Time to Buy and Sell Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int minV = Integer.MAX_VALUE, maxV = Integer.MIN_VALUE;
        if (prices.length==0){
            return 0;
        }
        for (int i = 0; i < prices.length; i++){
            minV = Math.min(minV,prices[i]);
            maxV = Math.max(maxV,prices[i]-minV);
        }
        return maxV;
    }
}